specification/dsdl/serializable_types.tex

```diff
@@ -281,51 +314,290 @@ \subsubsection{Dependencies}
+
+As detailed in section \ref{sec:dsdl_versioning},
+data types may evolve over time to accommodate new design requirements, new features, to rectify issues, etc.
+In order to allow gradual migration of deployed systems to revised data types,
+it is desirable to ensure that they can be modified in a way that does not render new definitions
+incompatible with their earlier versions.
+In this context there are two related concepts:
+
+\begin{description}
+    \item[Extent] --- the minimum amount of memory, in bits, that shall be allocated to store the serialized
+    representation of the type.
+    The extent of any type is always greater than or equal the maximal value of its bit length set.
+    It is always a multiple of 8.
+
+    \item[Sealing] --- a type that is \emph{sealed} is non-evolvable and its extent equals the maximal value
+    of its bit length set\footnote{%
+        I.e., the smallest possible extent.
+    }.
+    A type that is not sealed is also referred to as \emph{delimited}.
+\end{description}
+
+The extent is the growth limit for the maximal bit length of the type as it evolves.
+The extent should be at least as large as the longest serialized representation of any compatible version of the type,
+which ensures that an agent leveraging a particular version can correctly process any other compatible
+version due to the avoidance of unintended truncation of serialized representations.
+
+Serialized representations of evolvable definitions may carry additional metadata which introduces a certain overhead.
+This may be undesirable in some scenarios, especially in cases where serialized representations of the
+definition are expected to be highly compact, thereby making the overhead comparatively more significant.
+In such cases, the designer may opt out of the extensibility by declaring the definition sealed.
+Serialized representations of sealed definitions do not incur the aforementioned overhead.
+
+The related mechanics are described in section \ref{sec:dsdl_serialization_composite_non_sealed}.
+
+\begin{figure}[H]
+    $$
+    \overbrace{%
+        \underbrace{%
+            \blacksquare\blacksquare\blacksquare\blacksquare\blacksquare\blacksquare%
+            \blacksquare\blacksquare\blacksquare\blacksquare\blacksquare\blacksquare%
+        }_{\substack{\text{Longest serialized} \\ \text{representation}}}%
+        \underbrace{%
+            \boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes\boxtimes%
+        }_{\substack{\text{Memory reserve} \\ \text{(none if sealed)}}}%
+    }^{\substack{%
+        \text{Extent} \\
+        \text{(memory requirement)}
+    }}
+    $$
+    \caption{Serialized representation and extent\label{fig:dsdl_extent}}
+\end{figure}
+
+The default extent of a non-sealed (delimited) definition is the following function
+of the cumulative bit length set\footnote{Section \ref{sec:dsdl_composite_alignment_cumulative_bls}.}
+of its fields $B$:
```

Comment:
```suggestion
Where a non-sealed (delimited) type is defined \emph{without} an explicit extent value a default memory reserve of 50\% is implied using the following function
of the cumulative bit length set\footnote{Section \ref{sec:dsdl_composite_alignment_cumulative_bls}.}
of its fields $B$:
```

Reply:
> a default memory reserve of 50\% is implied using the following function

I don't like the redundancy here. The 50% overhead is defined in the equation already, why introduce it twice?

Reply:
I updated my suggestion.